### PR TITLE
Reviewing public access modifiers to save further on the binary size.

### PR DESCRIPTION
**Pull request by rdeassis**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

As another iteration of the reduction of the Vnext binary size effort and following changes of the PRs #445 and #450, we're now revisiting all the library and ensuring that only the members that need to be public or internal are kept that way.

We're default everything to private, internal, public/open in that order of necessity.

In the same metrics used in the change merged in PR #450 (fat binary including x86_64 and arm64 slices), changes on the access modifiers **decreased the size footprint caused by Vnext additions were cut by about 7%: (7.9MB to 7.3MB).** 

**Edit:** A significant merge from the main branch (through the vnext-prototype branch) into this PR branch increased the overall static lib size of FluentUI. New screenshots were attached after the merge happen to illustrate that the savings of **~0.6MB** still apply after the changes on this PR.
### Verification

**Before merge from main (through vnext-prototype):**
| Before changes on this PR                                |  After changes on this PR                               |
|----------------------------------------------|--------------------------------------------|
| Release_x86_64_arm64_before | Release_x86_64_arm64_after |

**After merge from main (through vnext-prototype):**
| Before changes on this PR                                |  After changes on this PR                               |
|----------------------------------------------|--------------------------------------------|
| Release_x86_64_arm64_merged_before | Release_x86_64_arm64_merged_after |

1. pod lib lint on both podspecs in the vnext-prototype branch
2. Sanity check by running Demo app
3. Building static library to validate size differences

### Pull request checklist

This PR has considered:
- [ ] Light and Dark appearances
- [ ] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [ ] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/453)

### CODE REVIEW

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarLegacyDemoController.swift [hunk 1]
```diff
@@ -165,7 +165,7 @@ class AvatarLegacyViewDemoController: DemoController {
 
         let avatarContainer = UIView()
         avatarContainer.addSubview(avatarView)
-        avatarContainer.widthAnchor.constraint(equalToConstant: AvatarLegacySize.extraExtraLarge.size.width).isActive = true
+        avatarContainer.widthAnchor.constraint(equalToConstant: 72).isActive = true
```
Comment (sophialee0416) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarLegacyDemoController.swift:168]: Why not keep it to what it was before?

Comment (rdeassis) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarLegacyDemoController.swift:168]: Keeping the enum size var public will make the binary larger. I think that the demo app is not enough to expose it.

File: ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift [hunk 2]
```diff
@@ -50,7 +50,7 @@ class DemoListViewController: UITableViewController {
         // Fluent UI design recommends not showing "Back" title. However, VoiceOver still correctly says "Back" even if the title is hidden.
         navigationItem.backBarButtonItem = UIBarButtonItem(title: "", style: .plain, target: nil, action: nil)
 
-        tableView.backgroundColor = Colors.Table.background
+        tableView.backgroundColor = Colors.tableBackground
```
Comment (sophialee0416) [inline ios/FluentUI.Demo/FluentUI.Demo/DemoListViewController.swift:53]: Do we want to change [this](https://github.com/microsoft/fluentui-apple/blob/d16c8091a198293aacc893fb73e1d931e358b42b/ios/FluentUI/People%20Picker/PersonaListView.swift#L90) as well?

**Review by sophialee0416 — APPROVED** [review]

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarLegacyDemoController.swift (hunk 1 above)
Comment (rdeassis) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarLegacyDemoController.swift:168]: Actually I found some uses of it in devmain. I'll revert this particular change.

Comment (sophialee0416) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/AvatarLegacyDemoController.swift]: Does the binary size increase with the reverted changes? If so maybe the description could be updated

**Review by sophialee0416 — APPROVED** [review]

**Comment by harrieshin** [comment]
we will see how many people gets made by making these private :)

**Review by harrieshin — APPROVED** [review]

**Pull request merged**